diff --git a/node_modules/@analogjs/vite-plugin-angular/src/lib/angular-vite-plugin.js b/node_modules/@analogjs/vite-plugin-angular/src/lib/angular-vite-plugin.js
index 83e6d13..39389f6 100644
--- a/node_modules/@analogjs/vite-plugin-angular/src/lib/angular-vite-plugin.js
+++ b/node_modules/@analogjs/vite-plugin-angular/src/lib/angular-vite-plugin.js
@@ -156,7 +156,7 @@ export function angular(options) {
                         if (req.url === undefined || res.writableEnded) {
                             return;
                         }
-                        if (!req.url.startsWith(ANGULAR_COMPONENT_PREFIX)) {
+                        if (!req.url.includes(ANGULAR_COMPONENT_PREFIX)) {
                             next();
                             return;
                         }
@@ -325,7 +325,7 @@ export function angular(options) {
                 }
                 if (pluginOptions.liveReload &&
                     options?.ssr &&
-                    id.startsWith(ANGULAR_COMPONENT_PREFIX)) {
+                    id.includes(ANGULAR_COMPONENT_PREFIX)) {
                     const requestUrl = new URL(id.slice(1), 'http://localhost');
                     const componentId = requestUrl.searchParams.get('c');
                     if (!componentId) {
